pkg/performanceprofile/controller/performanceprofile/status/status.go

```diff
@@ -21,18 +21,18 @@ import (
 )
 
 const (
-	conditionFailedToFindMachineConfigPool   = "GettingMachineConfigPoolFailed"
-	conditionBadMachineConfigLabels          = "BadMachineConfigLabels"
-	conditionReasonComponentsCreationFailed  = "ComponentCreationFailed"
-	conditionReasonMCPDegraded               = "MCPDegraded"
-	conditionFailedGettingMCPStatus          = "GettingMCPStatusFailed"
-	conditionKubeletFailed                   = "KubeletConfig failure"
-	conditionFailedGettingKubeletStatus      = "GettingKubeletStatusFailed"
-	conditionReasonTunedDegraded             = "TunedProfileDegraded"
-	conditionFailedGettingTunedProfileStatus = "GettingTunedStatusFailed"
+	ConditionFailedToFindMachineConfigPool   = "GettingMachineConfigPoolFailed"
+	ConditionBadMachineConfigLabels          = "BadMachineConfigLabels"
+	ConditionReasonComponentsCreationFailed  = "ComponentCreationFailed"
+	ConditionReasonMCPDegraded               = "MCPDegraded"
+	ConditionFailedGettingMCPStatus          = "GettingMCPStatusFailed"
+	ConditionKubeletFailed                   = "KubeletConfig failure"
+	ConditionFailedGettingKubeletStatus      = "GettingKubeletStatusFailed"
+	ConditionReasonTunedDegraded             = "TunedProfileDegraded"
+	ConditionFailedGettingTunedProfileStatus = "GettingTunedStatusFailed"
 )
 
-func (r *PerformanceProfileReconciler) updateStatus(profile *performancev2.PerformanceProfile, conditions []conditionsv1.Condition) error {
+func UpdateStatus(ctx context.Context, client client.Client, profile *performancev2.PerformanceProfile, conditions []conditionsv1.Condition) error {
```

Comment:
likewise, we may want to add a trivial `struct` to factor `context` and `client` instead of always repeating them

Reply:
Same constrains as here: https://github.com/openshift/cluster-node-tuning-operator/pull/1032#discussion_r1569036144